### ISSUE TITLE
Models

### ISSUE THREAD

**Issue by jeremie-gauthier** [jeremie-gauthier/HyperTube#17]
## User
| Field | Description | Type | Required | Default |
| ------ | ------ | ------ | ------ | ------ |
| id | User ID | `string` | ✔️ | None |
| username | The username of the user on the site | `string` | ✔️ | None |
| firstname | The user's firstname | `string` | ✔️ | None |
| lastname | The user's lastname | `string` | ✔️ | None |
| email | The user's email to authenticate | `string` | ❔ | None |
| password | The user's password to authenticate | `string` | ❔ | None |
| picture | The user's profile picture | `number` | ❌ | 1 |
| language | The user's default language setting | "en" \| "es" \| "fr" \| "ja" | ❌ | "en" |

*Note for user model:*
*email and password are probably not required if the auth is done with OAuth*

## Movie
| Field | Description | Type | Required | Default |
| ------ | ------ | ------ | ------ | ------ |
| id | User ID | `string` | ✔️ | None |

*send me your suggestions*

**Comment by jeremie-gauthier** [comment]
Send your suggestions in this conversation, I'll update my comment to add yours

**Comment by 0auBSQ** [comment]
We can keep the language in the local storage, no need to store it within the database

**Comment by jeremie-gauthier** [comment]
> We can keep the language in the local storage, no need to store it within the database

Not sure it's a good practice, because it won't sync user settings accross devices

**Comment by caro-rokah** [comment]
> We can keep the language in the local storage, no need to store it within the database

No, we need it in the DB. It is one of the informations that appear in the public profile of a user. 
So another user should have access to it too.
